msal/src/Microsoft.Identity.Client/Platforms/Android/AuthenticationContinuationHelper.cs

```diff
@@ -56,25 +56,51 @@ public static void SetAuthenticationContinuationEventArgs(int requestCode, Resul
             requestContext.Logger.InfoPii(msg);
             AuthorizationResult authorizationResult = null;
 
-            switch ((int) resultCode)
+            int code = (int)resultCode;
+
+            if (data.Action!=null && data.Action.Equals("ReturnFromEmbeddedWebview"))
+            {
+                authorizationResult = ProcessFromEmbeddedWebview(requestCode, resultCode, data);
+            } else
+            {
+                authorizationResult = ProcessFromSystemWebview(requestCode, resultCode, data);
+            } 
+            
+            WebviewBase.SetAuthorizationResult(authorizationResult, requestContext);
+        }
+
+        private static AuthorizationResult ProcessFromEmbeddedWebview(int requestCode, Result resultCode, Intent data)
```

Comment:
The reason for two is because in AuthenticationAgentActivity, the Intent before had the string "Return" which was getting registered as an `AuthorizationStatus.UnknownError`, so i changed it to contain the string "ReturnFromEmbeddedWebview", and then based on that result, will either process the authResult from system webview or embedded webview. I will look into streamlining this for another PR though.